R/latent_individual.R

```diff
@@ -138,94 +187,49 @@ epidist_formula.epidist_latent_individual <- function(data, delay_central = ~ 1,
       )
     )
   }
-  delay_equation <- paste0(
-    "delay_central | vreal(obs_t, pwindow_upr, swindow_upr)",
-    paste(delay_central, collapse = " ")
-  )
-  sigma_equation <- paste0("sigma", paste(sigma, collapse = " "))
-  form <- brms::bf(as.formula(delay_equation), as.formula(sigma_equation))
-  return(form)
-}
-
-#' @importFrom rstan lookup
-#' @method epidist_family epidist_latent_individual
-#' @family latent_individual
-#' @export
-epidist_family.epidist_latent_individual <- function(data, family = "lognormal",
-                                                     ...) {
-  epidist_validate(data)
-  checkmate::assert_string(family)
-
-  pdf_lookup <- rstan::lookup("pdf")
-  valid_pdfs <- gsub("_lpdf", "", pdf_lookup$StanFunction)
-  if (!family %in% valid_pdfs) {
-    cli::cli_warn(
-      "The provided family {.code family} does not correspond to a valid LPDF
-      function available in rstan. (It is possible [but unlikely] that there is
-      such an LPDF in Stan available via cmdstanr, as rstan is behind Stan.)"
-    )
-  }
-
-  brms::custom_family(
-    paste0("latent_", family),
-    dpars = c("mu", "sigma"),
-    links = c("identity", "log"),
-    lb = c(NA, 0),
-    ub = c(NA, NA),
-    type = "real",
-    vars = c("pwindow", "swindow", "vreal1"),
-    loop = FALSE
+  mu_index <- which(dpars == "mu")
```

Comment:
this one line? Anything we can do to avoid this? I see we need to know which one it is to inject the delay stuff but is there anything else we can do to identify it.

Reply:
Yes I also think it's a bit ugly. I'm not sure what about it I think is ugly though (so as to go about fixing it). I'm sure there are other ways. Maybe the thing that is ugly is that it relies on `dpars` having the same order as the formula input. They do because `dpars` was just extracted from the formula input, but you don't know that just looking at this line.

Reply:
`formula` is currently like

```
list(mu ~ 1, sigma ~ 1)
```

Hence it's quite hard to actually know which part is `mu` aside from extracting out the LHS (which is what I do to get `dpars`). We could require that it's `list("mu" = mu ~ 1, "sigma" = sigma ~ 1). Or I could use the `dpars` to name the list and then access them that way internally.